io/io/src/TBufferFile.cxx

```diff
@@ -1937,10 +1944,13 @@ void TBufferFile::WriteArrayDouble32(const Double_t *d, Int_t n, TStreamerElemen
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Write array of n bools into the I/O buffer.
+/// \note Operation is silently skipped in case of underflow or overflow
 
 void TBufferFile::WriteFastArray(const Bool_t *b, Int_t n)
 {
-   if (n <= 0) return;
+   constexpr Int_t dataWidth = static_cast<Int_t>(sizeof(UChar_t));
+   const Int_t maxElements = (std::numeric_limits<Int_t>::max() - Length())/dataWidth;
+   if (n <= 0 || n > maxElements) return;
```

Comment:
```suggestion
   if (n <= 0 || n > maxElements)
   {
      Fatal("WriteFastArray", "Not enough space left in the buffer (1GB limit).  %d elements is greater than the max left of %d", n, maxElements);
      return; // In case the user re-route the error handler to not die when Fatal is called)
   }
```